bastionzero/target/dbtarget/resource_dbtarget_test.go

```diff
@@ -0,0 +1,1133 @@
+					testAccCheckDbTargetExists(resourceName, &target),
+					testAccCheckDbTargetAttributes(t, &target, &expectedDbTarget{
+						EnvironmentID: &env.ID,
+						Name:          &rName,
+						ProxyTargetID: &bzeroTarget.ID,
+						RemoteHost:    bastionzero.PtrTo("localhost"),
+						RemotePort:    bastionzero.PtrTo(5432),
+						LocalPort:     nil,
+					}),
+					testAccCheckResourceDbTargetComputedAttr(resourceName),
+					resource.TestCheckNoResourceAttr(resourceName, "local_port"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccDbTarget_AllSupportedDatabaseAuthConfig(t *testing.T) {
+	// Test creating a db_target resource for each supported database
+	// authentication config returned by BastionZero.
+	//
+	// This test should fail if we add a new supported auth config and forget to
+	// update the Terraform provider's validation checks accordingly
+
+	ctx := context.Background()
+	rName := acctest.RandomName()
+	resourceName := "bastionzero_db_target.test"
+
+	acctest.SkipIfNotInAcceptanceTestMode(t)
+	acctest.PreCheck(ctx, t)
+
+	env := new(environments.Environment)
+	bzeroTarget := new(targets.BzeroTarget)
+	acctest.FindNEnvironmentsOrSkip(t, env)
+	acctest.FindNBzeroTargetsOrSkip(t, bzeroTarget)
+
+	// Get all currently supported db auth configs
+	supportedDbConfigs, _, err := acctest.APIClient.Targets.ListDatabaseAuthenticationConfigs(ctx)
+	if err != nil {
+		t.Fatalf("failed to list supported database auth configs: %s", err)
+	}
+
+	// Get valid remote_host based on database_authentication_config
+	validRemoteHost := func(authConfig *dbauthconfig.DatabaseAuthenticationConfig) string {
+		if authConfig.CloudServiceProvider != nil && *authConfig.CloudServiceProvider == dbauthconfig.GCP {
+			return "gcp://localhost"
+		} else if authConfig.CloudServiceProvider != nil && *authConfig.CloudServiceProvider == dbauthconfig.AWS && authConfig.Database != nil && *authConfig.Database == dbauthconfig.MySQL {
+			return "rdsmysql://localhost"
+		} else if authConfig.CloudServiceProvider != nil && *authConfig.CloudServiceProvider == dbauthconfig.AWS && authConfig.Database != nil && *authConfig.Database == dbauthconfig.Postgres {
+			return "rds://localhost"
+		} else {
+			return "localhost"
+		}
+	}
+
+	// Create TF config per db_auth_config. First step creates the resource with
+	// the first returned supported db_auth_config. Then each subsequent step
+	// updates the TF resource to the next supported db_auth_config, changing
+	// the remote_host if needed
+	steps := []resource.TestStep{}
+	for _, supportedConfig := range supportedDbConfigs {
+		var target targets.DatabaseTarget
+
+		// Why? See: https://go.dev/doc/faq#closures_and_goroutines
+		//
+		// Go language issue not fixed until Go 1.22: "This is because each
+		// iteration of the loop uses the same instance of the variable v, so
+		// each closure shares that single variable."
+		supportedConfig := supportedConfig // create a new 'supportedConfig' to bind in closure below.
```

Comment:
Interesting TIL but makes sense it would reuse the same variable. I would have named the variable something else but here you essentially hide the outer scoped variable right.

Reply:
Fixed here: 698e04f.

But yes it hides/shadows the outer scoped variable in the previous code